src/app/project/NewFeatureWizard.js

```diff
@@ -0,0 +1,282 @@
+            console.log('app.project.NewFeatureWizard:onFeatureCategoryChange', arguments);
+
+            this.resetFeatureAttributes();
+
+            this.graphicsLayer.clear();
+
+            var newValue = this.featureCategorySelect.value;
+            var toggle = (newValue === '') ? 'hide' : 'show';
+            $(this.geometryBtnsDiv).collapse(toggle);
+
+            if (toggle === 'hide') {
+                $(this.uploadBtn).collapse(toggle);
+            }
+
+            var items = config.domains.featureAttributes[newValue];
+            if (Array.isArray(items)) {
+                // line or point feature types
+                loadItemsIntoSelect(config.domains.pointLineActions, this.pointLineActionSelect);
+                if (items.length > 0) {
+                    loadItemsIntoSelect(items, this.typeSelect);
+                    domClass.remove(this.typeSelect.parentElement, 'hidden');
+                } else {
+                    domClass.add(this.typeSelect.parentElement, 'hidden');
+                }
+
+                domClass.remove(this.pointLineActionSelect.parentElement, 'hidden');
+            } else {
+                // polygon actions
+                var actions = [];
+                for (var action in items) {
+                    if (items.hasOwnProperty(action)) {
+                        actions.push(action);
+                    }
+                }
+                loadItemsIntoSelect(actions, this.polyActionSelect);
+                domClass.remove(this.polyActionSelect.parentElement, 'hidden');
+                domClass.remove(this.treatmentSelect.parentElement, 'hidden');
+            }
+        },
+        onUploadClick: function () {
+            // summary:
+            //      upload button was clicked
+            console.log('app.project.NewFeatureWizard:onUploadClick', arguments);
+
+            $(this.uploadDiv).collapse('show');
+        },
+        onFileSelected: function () {
+            // summary:
+            //      update controls after a file is selected for upload
+            console.log('app.project.NewFeatureWizard:onFileSelected', arguments);
+
+            this.uploadTxtBox.value = this.fileUploadInput.value.replace(/.*\\/, '');
+            this.uploadBtn.disabled = false;
+            $(this.uploadBtn).button('reset');
+        },
+        onUpload: function () {
+            // summary:
+            //      uploads the zip file
+            console.log('app.project.NewFeatureWizard:onUpload', arguments);
+
+            this.graphicsLayer.clear();
+            $(this.uploadBtn).button('loading');
+
+            var onError = function (er) {
+                topic.publish(config.topics.toast, {
+                    message: that.getServiceErrorMessage(er),
+                    type: 'danger'
+                });
+                $(that.uploadBtn).button('reset');
+            };
+
+            var that = this;
+            // upload file
+            esriRequest({
```

Comment:
I was told by an ESRI dev at the dev summit that if you are interacting with ESRI services, it's better to use esri/request since it does more checking than dojo/request.